globals/matchmaking_utils.go

```diff
@@ -140,6 +142,33 @@ func RemoveClientFromAllSessions(client *nex.Client) {
 	}
 }
 
+// CreateSessionByMatchmakeSession creates a gathering from a MatchmakeSession
+func CreateSessionByMatchmakeSession(matchmakeSession *match_making_types.MatchmakeSession, searchMatchmakeSession *match_making_types.MatchmakeSession, hostPID uint32) (*CommonMatchmakeSession, error, uint32) {
+	sessionIndex := GetAvailableGatheringID()
+	// This should in theory be impossible, as there aren't enough PIDs creating sessions to fill the uint32 limit.
+	// If we ever get here, we must be not deleting sessions properly
+	if sessionIndex == 0 {
+		return &CommonMatchmakeSession{}, fmt.Errorf("No gatherings available!"), nex.Errors.RendezVous.LimitExceeded
+	}
```

Comment:
Not an "issue" per say, but since we're already here refactoring stuff. I've been thinking about this, maybe we should just reset the counter instead of erroring here? I think it's a pretty safe assumption that if the uint32 limit is hit then the oldest sessions are very much dead (hell maybe even the people who made them are too)